Add new Rails/UnusedRenderContent cop

samrjenkins opened the pull request:
Issue: https://github.com/rubocop/rubocop-rails/issues/967

The Rails documentation states:
> If you try to render content along with a non-content status code (100-199, 204, 205, or 304), it will be dropped from the response.

https://guides.rubyonrails.org/layouts_and_rendering.html

On a number of occasions I have seen controller actions containing code such as:
```rb
render status: 204, plain: 'foo'
```
In cases such as these, the specified content for the response body is ignored and the code should be refactored to:
```rb
render status: 204
```

This pull request introduces a cop which identifies instances of unused body content.

-----------------

Before submitting the PR make sure the following are checked:

* [X] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [X] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [X] Feature branch is up-to-date with `master` (if not - rebase it).
* [X] Squashed related commits together.
* [X] Added tests.
* [X] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop-rails/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.
* [x] If this is a new cop, consider making a corresponding update to the [Rails Style Guide](https://github.com/rubocop/rails-style-guide).

[1]: https://chris.beams.io/posts/git-commit/

koic commented on config/default.yml:
```diff
@@ -1096,6 +1096,11 @@ Rails/UnusedIgnoredColumns:
 
+Rails/UnusedRenderContent:
+  Description: 'Do not specify body content for a response with a non-content status code.'
+  Enabled: pending
```
Can you add `Severity: warning` because this looks like a Lint-equivalent rule.

```suggestion
  Severity: warning
```

koic commented on lib/rubocop/cop/rails/unused_render_content.rb:
```diff
@@ -0,0 +1,68 @@
+      # If you try to render content along with a non-content status code (100-199, 204, 205, or 304),
+      # it will be dropped from the response.
+      #
+      # This cop checks for uses of `render` which specify both body content and a non-content status.
+
+      # @example
```
`#` required.

```suggestion
      # This cop checks for uses of `render` which specify both body content and a non-content status.
      #
      # @example
```

samrjenkins commented:
Thanks for your feedback, @koic. I have addressed both comments. I also rebased onto latest `master` to address issues of failing tests. Let me know if you would like me to squash the new commit.

samrjenkins commented:
@koic done 👍

koic commented:
Thanks!
